code/modules/client/preferences.dm

```diff
@@ -186,26 +186,29 @@ var/const/MAX_SAVE_SLOTS = 8
 		var/theckey = C.ckey
 		var/thekey = C.key
 		spawn()
-			while(!speciesinit)
-				sleep(1)
 			if(!IsGuestKey(thekey))
 				var/load_pref = load_preferences_sqlite(theckey)
 				if(load_pref)
-					if(!C)
-						WARNING("[__LINE__]: client C is null for [theckey].")
-					if(load_save_sqlite(theckey, C, default_slot) && C)
-						saveloaded = 1
-						return
-					else
-						world.log << "[theckey] failed loading save slot."
-				else
-					world.log << "[theckey] failed loading preferences."
+					while(!speciesinit)
```

Comment:
What's going on with the indentation here? The while block doesn't have anything inside it
Also if you put a while inside a while you will end up with double loops which is not great, maybe make try_load_save_sqlite not a while loop if you want a while loop in this block